buffer/src/main/java/io/netty/buffer/ByteBufAllocatorMetricProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer;
+
+public interface ByteBufAllocatorMetricProvider {
```

Comment:
>  I liked more to mark the "old methods" as @Deprecated

I see ... I wonder if this is necessary though as it also creates more work later to clean up these APIs. Is it a problem for the allocator to also provide its metrics ... seems natural, no?

>  not add a type param to the interface

Doesn't it make sense in this case? Presumably the Unpooled and Pooled allocator (and custom allocator X) will want to expose different types of metrics, no?